source/internal/_assertPromise.js

```diff
@@ -0,0 +1,8 @@
+import _isFunction from './_isFunction';
+import _toString from './_toString';
+
+export default function _assertPromise(name, p) {
```

Comment:
I read in earlier issues from the core devs (sorry, don't have references on hand) that Ramda's philosophy regarding undefined behavior was not to throw exceptions of your own but let low-level failures bubble up when users do something that doesn't make sense.

Is that a fair characterization, and if so, what makes this case different?

Reply:
@evan-king, I think you have a valid point. It will throw a type error anyways when trying to call the then method on the object.

```js
a.then()
// ReferenceError: a is not defined

a = {}; a.then()
//TypeError: a.then is not a function
```
By using JS default behavior we get more information about the reason of the error.